cSploit/src/org/csploit/android/MainActivity.java

```diff
@@ -887,16 +888,20 @@ public View getView(int position, View convertView, ViewGroup parent) {
     }
 
     public void clearSelection() {
-      for (Target t : list)
-        t.setSelected(false);
+      synchronized (this) {
+        for (Target t : list)
+          t.setSelected(false);
+      }
       notifyDataSetChanged();
```

Comment:
It is not mandatory to use **notifyDataSetChanged** always. if you want to update only one item in the list then you can use **notifyItemChanged(position)**